openstack/content-repo/bin/sleep-until-midnight

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+set -euo pipefail
+
+function sleepuntil ()
+{
+  current_epoch=$(date +%s)
+  target_epoch=$(date -d $1 +%s)
+
+  sleep_seconds=$(( $target_epoch - $current_epoch ))
+  #If time is already reached, wait for the next day
+  if [ ${sleep_seconds} -lt 0 ]; then
+    sleep_seconds=$((86400 + sleep_seconds))
+  fi
+
+  echo "Sleeping for $sleep_seconds seconds..."
+  sleep $sleep_seconds
+}
+
+# Run forever
+while :; do
+  # Run the command immediately
+  # Run until completed succesful
+  until "$@"; do :; done
```

Comment:
I'd like to have some backoff here, just in case that the command fails fast (e.g. with a config file error).